### PR TITLE
test(robot): migrate test_basic.py::test_restore_inc

### PR DESCRIPTION
**Pull request by yangchiu**
#### Which issue(s) this PR fixes:

Issue https://github.com/longhorn/longhorn/issues/8259

#### What this PR does / why we need it:

migrate test_basic.py::test_restore_inc

### CODE REVIEW

File: e2e/libs/keywords/volume_keywords.py [hunk 1]
```diff
@@ -24,9 +24,9 @@ def cleanup_volumes(self):
         for volume in volumes['items']:
             self.delete_volume(volume['metadata']['name'])
 
-    def create_volume(self, volume_name, size=2, replica_count=3, frontend="blockdev", migratable=False, access_mode="RWO", data_engine="v1", backing_image=""):
+    def create_volume(self, volume_name, size=2, numberOfReplicas=3, frontend="blockdev", migratable=False, accessMode="RWO", dataEngine="v1", backingImage="", Standby=False, fromBackup=""):
```
Comment (roger-ryao) [inline e2e/libs/keywords/volume_keywords.py:27]: I have a question. Why don't we continue to use snake_case for parameters? 
Are there any concerns?

Comment (yangchiu) [reply e2e/libs/keywords/volume_keywords.py:27]: I try to make the naming exactly identical to the longhorn volume crd, so users can set up a test case just like they would configure a Longhorn volume, without having to worry about the differences between Longhorn and Longhorn tests.

**Comment by yangchiu** [comment]
cc @longhorn/qa

File: e2e/tests/regression/test_backup.robot [hunk 2]
```diff
@@ -44,3 +49,58 @@ Test Backup Volume List
...
+    And Wait for volume 1 restoration from backup 0 completed
+    And Create DR volume 2 from backup 0
+    And Wait for volume 2 restoration from backup 0 completed
+    And Create DR volume 3 from backup 0
+    And Wait for volume 3 restoration from backup 0 completed
+
+    # Verify snapshot creation and PV/PVC creation are not allowed on a DR volume
+    Then Create snapshot 0 of volume 1 will fail
+    And Create persistentvolume for volume 1 will fail
+    And Create persistentvolumeclaim for volume 1 will fail
```
Comment (roger-ryao) [inline e2e/tests/regression/test_backup.robot:71]: One keyword that might help us better understand the story in the test report.
```
*** Keywords ***
Snapshot PV PVC could not be created on DR volume 1
    Create snapshot 0 of volume 1 will fail
    Create persistentvolume for volume 1 will fail
    Create persistentvolumeclaim for volume 1 will fail

Backup target could not be changed when DR volume exist
    Set setting backup-target to random.backup.target will fail


*** Test Cases ***
...
    And Wait for volume 3 restoration from backup 0 completed
    
    Then Snapshot PV PVC could not be created on DR volume 1
    And Backup target could not be changed when DR volume exist

    When Activate DR volume 1
...
```

Screenshot_20240607_161752

Comment (yangchiu) [reply e2e/tests/regression/test_backup.robot]: Refined. Thank you!

**Review by roger-ryao — APPROVED** [review]
LGTM

**Pull request merged**